R/stat-density-2d.r

```diff
@@ -60,30 +115,80 @@ StatDensity2d <- ggproto("StatDensity2d", Stat,
 
   required_aes = c("x", "y"),
 
+  extra_params = c(
+    "na.rm", "contour", "contour_var",
+    "bins", "binwidth", "breaks"
+  ),
+
+  # stat used for contouring
+  contour_stat = StatContour,
```

Comment:
I don't know whether I like that a stat contains another stat as a field. Why not just have it as the equivalent of the old parameter (i.e. a character string) and do the switch as you did before?

My main gripe was that this switch was user accessible, not that it was being done

Reply:
fixed.